docs/user-guide/federation/federated-services.md

```diff
@@ -0,0 +1,234 @@
+
+## Discovering a Federated Service
+
+### From Pods Inside your Federated Clusters
+
+By default, Kubernetes clusters come pre-configured with a cluster-local DNS server ('KubeDNS'), as well as an intelligently constructed DNS search path which together ensure that DNS queries like "myservice", "myservice.mynamespace", "bobsservice.othernamespace" etc issued by your software running inside Pods are automatically expanded and resolved correctly to the appropriate service IP of services running in the local cluster.
+
+With the introduction of Federated Services and Cross-Cluster Service Discovery, this concept is extended to cover Kubernetes services running in any other cluster across your Cluster Federation, globally.  To take advantage of this extended range, you use a slightly different DNS name (of the form "<servicename>.<namespace>.<federationname>", e.g. myservice.mynamespace.myfederation) to resolve Federated Services.  Using a different DNS name also avoids having your existing applications accidentally traversing cross-zone or cross-region networks and you incurring perhaps unwanted network charges or latency, without you explicitly opting in to this behavior.
+
+So, using our NGINX example service above, and the Federated Service DNS name form just described, let's consider an example: A Pod in a cluster in the `us-central1-f` availability zone needs to contact our NGINX service.  Rather than use the service's traditional cluster-local DNS name (```"nginx.mynamespace"```, which is automatically expanded to ```"nginx.mynamespace.svc.cluster.local"```) it can now use the service's Federated DNS name, which is ```"nginx.mynamespace.myfederation"```. This will be automatically expanded and resolved to the closest healthy shard of my NGINX service, wherever in the world that may be. If a healthy shard exists in the local cluster, that service's cluster-local (typically 10.x.y.z) IP address will be returned (by the cluster-local KubeDNS).  This is exactly equivalent to non-federated service resolution.
+
+But if the service does not exist in the local cluster (or it exists but has no healthy backend pods), the DNS query is automatically expanded to ```"nginx.mynamespace.myfederation.svc.us-central1-f.example.com"`` (i.e. logically "find the external IP of one of the shards closest to my availability zone").  This expansion is performed automatically by KubeDNS, which returns the associated CNAME record.  This results in automitic traversal of the hierarchy of DNS records in the above example, and ends up at one of the external IP's of the Federated Service in the local us-central1 region (i.e. 104.197.247.191, 104.197.244.180 or 104.197.245.170).
```

Comment:
> "nginx.mynamespace.myfederation.svc.us-central1-f.example.com"

Three closing back ticks instead of two here. Causing rendering problems right now.